### PR TITLE
Fix for issue - #225 (Create specific logs for each InstaPy file)

### PR DESCRIPTION
**Pull request by CodeMaster1**
This change is to create log files based on the account's username from
which the program is run. It creates three files based on username -
logfile_{username}.txt, followRestriction_{username}.json,
followed_{username}.txt.

### CODE REVIEW

**Comment by timgrossmann** [comment]
@CodeMaster1 @littleswany I actually think this is a V1 feature...
This will break the current versions feature of following only a defined number of times...

**Comment by CodeMaster1** [comment]
@timgrossmann So, you think we should not commit to this version's branch or should we copy existing followrestriction.json to followrestriction_{username}.json to commit feature in this branch?

**Comment by pekapa** [comment]
I think it would be better to work this as a v1 as well.
@CodeMaster1 could you try to implement it based on the dev branch? I think it still needs some work on the follow feature (that is undone at the moment) but it could be re-written in order to accommodate this changes.

File: instapy/instapy.py [hunk 1]
```diff
@@ -37,23 +37,33 @@ def __init__(self, username=None, password=None, nogui=False):
...
-    self.logFile = open('./logs/logFile.txt', 'a')
-    self.logFile.write('Session started - %s\n' \
-                       % (datetime.now().strftime('%Y-%m-%d %H:%M:%S')))
-
     self.username = username or environ.get('INSTA_USER')
     self.password = password or environ.get('INSTA_PW')
     self.nogui = nogui
 
-
+    logfile_name = './logs/logfile_' + self.username + '.txt'
```
Comment (timgrossmann) [inline instapy/instapy.py:44]: Please use python's formatted strings for this...e.g.  ''.format()

File: instapy/instapy.py [hunk 2]
```diff
@@ -37,23 +37,33 @@ def __init__(self, username=None, password=None, nogui=False):
...
     self.do_comment = False
     self.comment_percentage = 0
     self.comments = ['Cool!', 'Nice!', 'Looks good!']
     self.photo_comments = []
     self.video_comments = []
 
     self.followed = 0
-    self.follow_restrict = load_follow_restriction()
+
+    follow_res_json = './logs/followRestriction_' + self.username + '.json'
```
Comment (timgrossmann) [inline instapy/instapy.py:58]: Please use python's formatted strings for this...e.g.  ''.format()

File: instapy/instapy.py [hunk 3]
```diff
@@ -399,6 +409,7 @@ def end(self):
     )
     self.logFile.write('-' * 20 + '\n\n')
     self.logFile.close()
-
-    with open('./logs/followed.txt', 'w') as followFile:
+    
+    followed_logfile_name = './logs/followed_' + self.username + '.txt'
```
Comment (timgrossmann) [inline instapy/instapy.py:413]: Please use python's formatted strings for this...e.g.  ''.format()

File: instapy/print_log_writer.py [hunk 4]
```diff
@@ -8,5 +8,6 @@ def log_follower_num(browser, username):
   browser.get('https://www.instagram.com/' + username)
 
   followed_by = browser.execute_script("return window._sharedData.entry_data.ProfilePage[0].user.followed_by.count")
-  with open('./logs/followerNum.txt', 'a') as numFile:
+  follower_logfile_name = './logs/followerNum_' + username + '.txt'
```
Comment (timgrossmann) [inline instapy/print_log_writer.py:11]: Please use python's formatted strings for this...e.g.  ''.format()

File: instapy/unfollow_util.py [hunk 5]
```diff
@@ -69,12 +69,14 @@ def follow_given_user(browser, acc_to_follow, follow_restrict):
...
-  with open('./logs/followRestriction.json', 'w') as followResFile:
+  follow_res_json = './logs/followRestriction_' + username + '.json'
+  with open(follow_res_json, 'w') as followResFile:
     json.dump(followRes, followResFile)
 
-def load_follow_restriction():
+def load_follow_restriction(username):
   """Loads the saved """
-  with open('./logs/followRestriction.json') as followResFile:
+  follow_res_json = './logs/followRestriction_' + username + '.json'
```
Comment (timgrossmann) [inline instapy/unfollow_util.py:80]: Please use python's formatted strings for this...e.g.  ''.format()

File: instapy/unfollow_util.py [hunk 6]
```diff
@@ -69,12 +69,14 @@ def follow_given_user(browser, acc_to_follow, follow_restrict):
         sleep(3)
         return  0
 
-def dump_follow_restriction(followRes):
+def dump_follow_restriction(followRes, username):
   """Dumps the given dictionary to a file using the json format"""
-  with open('./logs/followRestriction.json', 'w') as followResFile:
+  follow_res_json = './logs/followRestriction_' + username + '.json'
```
Comment (timgrossmann) [inline instapy/unfollow_util.py:74]: Please use python's formatted strings for this...e.g.  ''.format()

**Review by timgrossmann — CHANGES_REQUESTED** [review]

**Comment by CodeMaster1** [comment]
I'll add the changes to the latest branch,

**Pull request closed without merging**